src/generic/FreeModule.jl

```diff
@@ -10,15 +10,15 @@
 #
 ###############################################################################
 
-parent_type(::Type{FreeModuleElem{T}}) where T <: Union{RingElement, NCRingElem} = FreeModule{T}
+parent_type(::Type{FreeModuleElem{T}}) where T <: NCRingElement = FreeModule{T}
 
-base_ring_type(::Type{FreeModule{T}}) where T <: Union{RingElement, NCRingElem} = parent_type(T)
+base_ring_type(::Type{FreeModule{T}}) where T <: NCRingElement = parent_type(T)
 
-base_ring(M::FreeModule{T}) where T <: Union{RingElement, NCRingElem} = M.base_ring::parent_type(T)
+base_ring(M::FreeModule{T}) where T <: NCRingElement = M.base_ring::parent_type(T)
 
-elem_type(::Type{FreeModule{T}}) where T <: Union{RingElement, NCRingElem} = FreeModuleElem{T}
+elem_type(::Type{FreeModule{T}}) where T <: NCRingElement = FreeModuleElem{T}
 
-parent(m::FreeModuleElem{T}) where T <: Union{RingElement, NCRingElem} = m.parent
+parent(m::FreeModuleElem{T}) where T <: NCRingElement = m.parent
 
 function rels(M::FreeModule{T}) where T <: RingElement
```

Comment:
This is going beyond the scope of your PR, but I wonder, shouldn't this also be
```suggestion
function rels(M::FreeModule{T}) where T <: NCRingElement
```
or
```suggestion
function rels(M::FreeModule{T}) where T
```

Reply:
Yes, I would think so too.